testing/internal/e2e/tests/base/auth_unauthenticated_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package base_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCliUnauthenticatedUserAccess(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+
+	// Check one authenticated request
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"targets", "list",
+			"--recursive",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Logout
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"logout",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Attempt commands that unauthetnicated users can make
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"scopes", "list",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"auth-methods", "list",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// Check user cannot make requests that require authentication
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"targets", "list",
+			"--recursive",
+		),
+	)
+	require.Error(t, output.Err, "Unauthenticated user was able to access private resource")
+
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"sessions", "list",
+			"--recursive",
+		),
+	)
+	require.Error(t, output.Err, "Unauthenticated user was able to access private resource")
+
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"users", "list",
+			"--recursive",
+		),
+	)
+	require.Error(t, output.Err, "Unauthenticated user was able to access private resource")
+}
```

Comment:
It would be interesting if this test (or maybe a separate test) could try changing the permissions for the unauthenticated user and see that they are reflected in what the user is able to do. Maybe a follow up?

Reply:
Yea. Let's do that in a separate ticket/PR.

I'm trying to play around with it locally, but I'm not sure I'm seeing the expected behavior